wordpress-mobile/WordPress-iOS
# Notifications > Comment Details: fetch comment if it is not cached

Ref: #17790 
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/486

When selecting a comment notification to view, if the comment is not cached, it is now fetched from the endpoint added in the referenced WPKit PR.

Notes:
- A loading view is not yet displayed, so sometimes there is a delay between tapping the notification row and the details appearing.
- An error view is not yet displayed, so if the comment can't be loaded nothing will happen when the notification row is tapped.

To test:
- Enable `notificationCommentDetails`.
- Start with a fresh install to ensure no comments are cached.
- Go to Notifications > Comments.
- Select any notification.
- Verify the new comment view is displayed.

| Before | After |
|--------|-------|
| ![before](https://user-images.githubusercontent.com/1816888/152438951-d9d64d01-e4d9-4a5b-95a8-6c898c406fe2.png) | ![after](https://user-images.githubusercontent.com/1816888/152438957-6fd42cab-e4bf-4b5f-bb8a-fa657c6ec223.png) |

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is disabled.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is disabled.

3. What automated tests I added (or what prevented me from doing so)
Tests added in WPKit.

PR submission checklist:
